ubuntu_image/parser.py

```diff
@@ -64,6 +65,16 @@ def __call__(self, v):
             ]
 
 
+def Size32bit(v):
```

Comment:
What do you think about giving `as_size()` optional min and max parameters, e.g.

```
def as_size(size, min=0, max=None)
```

Then `Size32bit()` could be implemented easily as `return as_size(v, max=GiB(4))`

We could then test the corner cases of `as_size()` in test_helpers.py.  `max` would be exclusive, e.g. the range would be [min, max)

I also wonder if it wouldn't just be easier to check the range at the end of `as_size()` by doing

```
if not (min <= converted_value < max):
    raise ValueError
```

We probably wouldn't need the struct module then.  What do you think?